src/dict.c

```diff
@@ -1074,12 +1079,34 @@ unsigned int dictGetSomeKeys(dict *d, dictEntry **des, unsigned int count) {
 }
 
 
-/* Reallocate the dictEntry allocations in a bucket using the provided
- * allocation function in order to defrag them. */
-static void dictDefragBucket(dict *d, dictEntry **bucketref, dictDefragAllocFunction *allocfn) {
-    while (bucketref && *bucketref && !entryIsKey(*bucketref)) {
-        dictEntry *de = *bucketref, *newde;
-        if ((newde = allocfn(de))) {
+/* Reallocate the dictEntry, key and value allocations in a bucket using the
+ * provided allocation functions in order to defrag them. */
+static void dictDefragBucket(dict *d, dictEntry **bucketref, dictDefragFunctions *defragfns) {
+    dictDefragAllocFunction *allocfn = defragfns->defragAlloc;
```

Comment:
```suggestion
    dictDefragAllocFunction *defragAlloc = defragfns->defragAlloc;
```
it's a defrag fn, not an alloc fn. also more consistent with the other two.